doc/toolchains/llvm.chapter.md

```diff
@@ -0,0 +1,31 @@
+# The LLVM Toolchain {#chap-toolchains}
+
+LLVM is a target-independent optimizer and code generator and serves as the basis for many compilers such as Haskell's GHC, Rust, Zig, and many others. It forms the base tools for Apple's Darwin platform.
+
+## Using LLVM {#sec-using-llvm}
+
+LLVM has two ways of being used. One is by using it across all of nixpkgs and the other is to compile and build other packages.
+
+### Building packages with LLVM {#sec-building-packages-with-llvm}
+
+Nixpkgs supports two methods of compiling the world with LLVM, one is via setting `useLLVM` in `crossSystem` while importing. This is the recommended way when doing cross compiling as it is more expressive. An example of doing `aarch64-linux` cross from `x86_64-linux` with LLVM on the target is by doing the following:
+
+```nix
+import <nixpkgs> {
+  localSystem = {
+    system = "x86_64-linux";
+  };
+  crossSystem = {
+    useLLVM = true;
+    linker = "lld";
+  };
+}
+```
+
+Note that we set `linker` to `lld`, this is because LLVM has its own linker called "lld". By setting it, we utilize Clang and lld within this new instance of nixpkgs. There is a shorthand method which nixpkgs supports for building everything with LLVM is via `pkgsLLVM`. This is easier to be used with `nix-build` (or `nix build`). An example of this is as simple as the following:
+
+```bash
+nix-build -A pkgsLLVM.hello
+```
+
+This will compile the GNU hello package with LLVM and the lld linker like previously mentioned. Another simple way is to override the stdenv with `clangStdenv`. This causes a single package to be built with Clang. However, this `stdenv` does not provide compiler-rt, libc++, and libunwind. It will use the platform's default. This is the preferred way to make something use Clang in nixpkgs when packaging a piece of software. There are cases where providing just Clang isn't enough, this is where the `libcxxStdenv` comes in. `libcxxStdenv` uses Clang with libc++ and compiler-rt.
```

Comment:
I think it would be good if `useLLVM` and `clangStdenv` were each other their own subheadings.  `clangStdenv` is really easy to miss here because the paragraph it's explained in starts by being about `useLLVM`.

Reply:
It has been split off into it's own heading now.